Specs/Core/loadKTX2Spec.js

```diff
@@ -0,0 +1,322 @@
+      32,
+      32,
+      true
+    );
+  });
+
+  it("transcodes ETC1S ktx2 to bc7", function () {
+    return expectKTX2TranscodeResult(
+      "./Data/Images/Green4x4_ETC1S.ktx2",
+      { bc7: true },
+      4,
+      4,
+      true
+    );
+  });
+
+  it("transcodes UASTC ktx2 to bc7", function () {
+    return expectKTX2TranscodeResult(
+      "./Data/Images/Logo32x32_UASTC_Zstd.ktx2",
+      { bc7: true },
+      32,
+      32,
+      true
+    );
+  });
+
+  it("returns a promise that resolves to an uncompressed texture", function () {
+    return expectKTX2TranscodeResult(
+      "./Data/Images/Green4x4.ktx2",
+      { s3tc: true },
+      4,
+      4,
+      false
+    );
+  });
+
+  it("returns a promise that resolves to an uncompressed texture containing all mip levels of the original texture", function () {
+    var resource = Resource.createIfNeeded("./Data/Images/Green4x4Mipmap.ktx2");
+    var loadPromise = resource.fetchArrayBuffer();
+    return loadPromise.then(function (buffer) {
+      var promise = KTX2Transcoder.transcode(buffer, {});
+      expect(promise).toBeDefined();
+      return promise.then(function (resolvedValue) {
+        expect(resolvedValue).toBeDefined();
+        expect(resolvedValue.length).toEqual(3);
+        var dims = [4, 2, 1];
+        for (var i = 0; i < resolvedValue.length; ++i) {
+          expect(resolvedValue[i].width).toEqual(dims[i]);
+          expect(resolvedValue[i].height).toEqual(dims[i]);
+          expect(
+            PixelFormat.isCompressedFormat(resolvedValue[i].internalFormat)
+          ).toEqual(false);
+          expect(resolvedValue[i].bufferView).toBeDefined();
+        }
+      });
+    });
+  });
+
+  it("returns a promise that resolves to a compressed texture containing the all mip levels of the original texture", function () {
+    var resource = Resource.createIfNeeded(
+      "./Data/Images/Green4x4Mipmap_ETC1S.ktx2"
+    );
+    var loadPromise = resource.fetchArrayBuffer();
+    return loadPromise.then(function (buffer) {
+      var promise = KTX2Transcoder.transcode(buffer, { etc1: true });
+      expect(promise).toBeDefined();
+      return promise.then(function (resolvedValue) {
+        expect(resolvedValue).toBeDefined();
+        expect(resolvedValue.length).toEqual(3);
+        var dims = [4, 2, 1];
+        for (var i = 0; i < resolvedValue.length; ++i) {
+          expect(resolvedValue[i].width).toEqual(dims[i]);
+          expect(resolvedValue[i].height).toEqual(dims[i]);
+          expect(
+            PixelFormat.isCompressedFormat(resolvedValue[i].internalFormat)
+          ).toEqual(true);
+          expect(resolvedValue[i].bufferView).toBeDefined();
+        }
+      });
+    });
+  });
+
+  it("cannot parse invalid KTX2 buffer", function () {
+    var invalidKTX = new Uint8Array([0, 1, 2, 3, 4, 5]);
+    var resolvedValue;
+    var rejectedError;
+    var promise = loadKTX2(invalidKTX.buffer);
+    expect(promise).toBeDefined();
```

Comment:
It's ok not to do `expect(promise).toBeDefined();` in these tests. The test will fail if it's undefined.